src/libopenrave/robotmanipulator.cpp

```diff
@@ -392,297 +392,202 @@ bool RobotBase::Manipulator::FindIKSolutions(const IkParameterization& goal, con
-            ikp.SetTranslationXAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.x,-1.0,1.0)));
-        }
-        else {
-            const Vector vlocaldirection = t.rotate(_info._vdirection);
-            ikp.SetTranslationXAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.x,-1.0,1.0)));
-        }
-        break;
-    }
-    case IKP_TranslationYAxisAngle4D: {
-        if (inworld) {
-            Transform localt = GetBase()->GetTransform().inverse()*t;
-            const Vector vlocaldirection = localt.rotate(_info._vdirection);
-            ikp.SetTranslationYAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.y,-1.0,1.0)));
-        }
-        else {
-            const Vector vlocaldirection = t.rotate(_info._vdirection);
-            ikp.SetTranslationYAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.y,-1.0,1.0)));
-        }
-        break;
-    }
-    case IKP_TranslationZAxisAngle4D: {
-        if (inworld) {
-            Transform localt = GetBase()->GetTransform().inverse()*t;
-            const Vector vlocaldirection = localt.rotate(_info._vdirection);
-            ikp.SetTranslationZAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.z,-1.0,1.0)));
-        }
-        else {
-            const Vector vlocaldirection = t.rotate(_info._vdirection);
-            ikp.SetTranslationZAxisAngle4D(t.trans,RaveAcos(utils::ClampOnRange(vlocaldirection.z,-1.0,1.0)));
-        }
-        break;
-    }
-    case IKP_TranslationXAxisAngleZNorm4D: {
-        if (inworld) {
-            Transform localt = GetBase()->GetTransform().inverse()*t;
-            const Vector vlocaldirection = localt.rotate(_info._vdirection);
-            ikp.SetTranslationXAxisAngleZNorm4D(t.trans,RaveAtan2(vlocaldirection.y,vlocaldirection.x));
-        }
-        else {
-            Vector vlocaldirection = t.rotate(_info._vdirection);
-            ikp.SetTranslationXAxisAngleZNorm4D(t.trans,RaveAtan2(vlocaldirection.y,vlocaldirection.x));
-        }
-        break;
-    }
-    case IKP_TranslationYAxisAngleXNorm4D: {
-        if (inworld) {
-            Transform localt = GetBase()->GetTransform().inverse()*t;
-            const Vector vlocaldirection = localt.rotate(_info._vdirection);
-            ikp.SetTranslationYAxisAngleXNorm4D(t.trans,RaveAtan2(vlocaldirection.z,vlocaldirection.y));
-        }
-        else {
-            Vector vlocaldirection = t.rotate(_info._vdirection);
-            ikp.SetTranslationYAxisAngleXNorm4D(t.trans,RaveAtan2(vlocaldirection.z,vlocaldirection.y));
-        }
-        break;
-    }
-    case IKP_TranslationZAxisAngleYNorm4D: {
-        if (inworld) {
-            Transform localt = GetBase()->GetTransform().inverse()*t;
-            const Vector vlocaldirection = localt.rotate(_info._vdirection);
-            ikp.SetTranslationZAxisAngleYNorm4D(t.trans,RaveAtan2(vlocaldirection.x,vlocaldirection.z));
+        if(std::abs(vdir.dot(vdir)) > 1e-12){
```

Comment:
could we improve this part by avoiding a magic number here?

Reply:
would you mind telling me global epsilon definition?

Reply:
How about using `g_fEpsilon`?

Reply:
so https://github.com/rdiankov/openrave/blob/production/include/openrave_c/openrave_c.h#L43 ok got it!

Reply:
done (sorry to be late, needed to check compile)